src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogAgentClient.java

```diff
@@ -282,24 +310,161 @@ private boolean reinitializeLogger(boolean force) {
         return true;
     }
 
+    /**
+     * Fetches the supported endpoints form the Trace Agent /info API
+     *
+     * @return a list of endpoints (if /info wasn't available, it will be empty)
+     */
+    @SuppressFBWarnings("REC_CATCH_EXCEPTION")
```

Comment:
Could we only call this when we know the agent is running?

Reply:
The whole `DatadogAgentClient` class is only used if Jenkins is configured to use the Agent, so I assume it will be present or otherwise we can consider it a configuration error.

Reply:
Sorry, I meant in this case:

> If Jenkins starts before the Agent, then the EVP Proxy support will be disabled because we won't be able to query the /info API when initializing the DatadogAgentClient.

I'm wondering if we could ensure we only attempt this when the agent is reachable, so we don't have to retry

Reply:
An alternative would be to lazy resolve the `/info` endpoint when it is first needed, ie: on the first job that runs. I'm not sure if that could create concurrency problems if two jobs run at the same time, though, so I went with the simpler approach.

I would be interested in hearing @drodriguezhdez's opinion now that he's back :)